src/libponyrt/ds/hash.c

```diff
@@ -22,13 +22,18 @@ static void* search(hashmap_t* map, size_t* pos, void* key, size_t hash,
   size_t index = ((hash & mask) + p_length) & mask;
   void* elem;
   size_t elem_hash;
+  size_t ib_index;
+  size_t ib_offset;
 
   for(size_t i = 0; i <= mask; i++)
   {
+    ib_index = index/HASHMAP_BITMAP_TYPE_SIZE;
+    ib_offset = index%HASHMAP_BITMAP_TYPE_SIZE;
```

Comment:
This can be optimised to bit-wise operations.

```
ib_index = index >> HASHMAP_BITMAP_TYPE_BITS;
ib_offset = index & HASHMAP_BITMAP_TYPE_MASK;
```

Where, in the header:

```
#define HASHMAP_BITMAP_TYPE_BITS 8
#define HASHMAP_BITMAP_TYPE_SIZE (1 << HASHMAP_BITMAP_TYPE_BITS)
#define HASHMAP_BITMAP_TYPE_MASK (HASHMAP_BITMAP_TYPE_SIZE - 1)
```